File: simple-web-server/src/main/java/de/thi/informatik/gdp2/simplewebapp/model/Room.java
package de.thi.informatik.gdp2.simplewebapp.model;

import java.util.LinkedList;
import java.util.List;
import java.util.stream.Collectors;

public class Room {
	private String name;
	private List<Message> list;

	public Room(String name) {
		this.name = name;
		this.list = new LinkedList<>();
	}
	
	public String getName() {
		return name;
	}
	
	public void addMessage(String user, String message) {
		this.list.add(new Message(user, message));
	}
	
	public List<Message> getMessagesSince(long time) {
		return this.list.stream().filter(msg -> msg.getTime() >= time).collect(Collectors.toList());
	}
}

File: simple-web-server-kotlin/src/main/kotlin/de/thi/informatik/gdp2/simplewebserverkotlin/model/Room.kt
package de.thi.informatik.gdp2.simplewebserverkotlin.model

import java.util.stream.Collector
import java.util.stream.Collectors

class Room {
    val name: String
        get() = field
    private val list: MutableList<Message>

    constructor(name: String) {
        this.name = name
        this.list = ArrayList<Message>()
    }

    fun addMessage(user: String, message: String) =
        this.list.add(Message(user, message))

    fun getMessagesSince(time: Long): List<Message> =
        this.list.stream().filter { msg -> msg.time >= time }.collect(Collectors.toList())
}